### PR TITLE
feat: file tree 构建方法实现

### LINKED ISSUE #71: Generator 类：files tree 构建方法实现
调用所有 plugin 的 generator 后，相关的文件可以从 template render 到内存，也可以从扩展后的 pka 中提取，经过相关加工后，写文件到项目中

### PR DESCRIPTION
**Pull request by IsDyh01**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/xun082/create-neat/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[✅] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

- Closes #71 
- Related to #24  

## What is the new behavior?
调用所有 plugin 的 generator 后，相关的文件可以从 template render 到内存，也可以从扩展后的 pka中提取，经过相关加工后，写文件到项目中.

## Does this PR introduce a breaking change?

```
[✅ ] Yes
[ ] No
```

### CODE REVIEW

File: packages/core/src/utils/ConfigTransform.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+export default class ConfigTransform {
+  private fileDesc; // 文件类型描述
+  constructor(fileDesc) {
+    this.fileDesc = fileDesc.file;
+  }
+
+  /**
+   *
```
Comment (uaenaTzx) [inline packages/core/src/utils/ConfigTransform.ts:8]: @descript 函数功能

File: packages/core/src/utils/Generator.ts [hunk 2]
```diff
@@ -3,12 +3,64 @@ import path from "path";
...
+  "lint-staged": new ConfigTransform({
+    file: {
+      js: ["lint-staged.config.js"],
+      json: [".lintstagedrc", ".lintstagedrc.json"],
+      yaml: [".lintstagedrc.yaml", ".lintstagedrc.yml"],
+    },
+  }),
+};
+
+// vue项目对应得配置文件
```
[the remark this answers is not in the document]
Comment (uaenaTzx) [reply packages/core/src/utils/Generator.ts:55]: 注释错别字

File: packages/core/src/utils/Generator.ts [hunk 3]
```diff
@@ -62,12 +128,51 @@ class Generator {
     // 整合需要安装的文件
     // 这里假设 GeneratorAPI 有一个方法来更新这个 Generator 实例的 files
     // createFiles 函数需要你根据自己的逻辑实现文件创建和写入磁盘的逻辑
+    // extract configs from package.json into dedicated files.
+    // 从package.json中生成额外得的文件
+    await this.extractConfigFiles();
+
+    // 重写pakcage.json文件，消除generatorAPI中拓展package.json带来得副作用
```
Comment (uaenaTzx) [inline packages/core/src/utils/Generator.ts:135]: 的 mark

File: packages/utils/src/core/getResolveApp.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import fs from "node:fs";
 import path from "node:path";
 
 const appDirectory: string = fs.realpathSync(process.cwd());
+console.log(appDirectory, "------appDirectory");
```
Comment (uaenaTzx) [inline packages/utils/src/core/getResolveApp.ts:5]: 日志去掉吧

File: packages/core/src/utils/ConfigTransform.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      filename,
+      content,
+    };
+  }
+
+  /**
+   * 获取文件类型和文件名
+   * @returns {type, filename} type为需要生成得文件类型，filename为文件名
+   */
+  getDefaultFile() {
```
Comment (qq1051766345) [inline packages/core/src/utils/ConfigTransform.ts:29]: 尽量把函数定义写在函数调用之前

File: packages/core/src/utils/createAppTest.ts [hunk 6]
```diff
@@ -107,12 +107,14 @@ export default async function createAppTest(projectName: string, options: Record
   await dependenciesInstall(rootDirectory, packageManager);
 
   // 运行生成器创建项目所需文件和结构
+
   console.log(chalk.blue(`🚀  Invoking generators...`));
-  const generators = new Generator(rootDirectory, plugins);
+  const generators = new Generator(rootDirectory, preset.plugins, packageContent);
```
Comment (uaenaTzx) [inline packages/core/src/utils/createAppTest.ts:112]: preset属性在前面已经解构出来了

**Pull request merged**